components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/AbstractApplicationAuthenticator.java

```diff
@@ -216,6 +301,9 @@ protected boolean retryAuthenticationEnabled() {
         return false;
     }
 
+    protected boolean eventFiringEnabled() {
```

Comment:
We can have a better name for this method. Have some add a java doc comment explaining what the logic behind this.